content/404.html.haml

```diff
@@ -19,3 +25,37 @@ notitle: true
   %p
   %a{:href => "/"} Back to homepage
 
+%div.searchcontainer
+  %h4
+    Search for related pages
+
+:javascript
+  // fetch last segment from URL
+  const path = window.location.pathname;
+  const pathSegments = path.split('/');
```

Comment:
https://deploy-preview-7026--jenkins-io-site-pr.netlify.app/404/

expected: the initial phrase of the new search dialog is "404"
actual: the search phrase is empty. 

This is because of the trailing slash that's added to URLs automatically.
```suggestion
  const pathSegments = path.split('/').filter(Boolean);
```
should fix it.